### PR TITLE
enhance: [GoSDK] Sync IndexType supported from v1

### PR DESCRIPTION
**Pull request by congqixia**
See also #31293

### CODE REVIEW

File: client/index/ivf.go [hunk 1]
```diff
@@ -106,3 +110,30 @@ func NewIvfSQ8Index(metricType MetricType, nlist int) Index {
...
+
+func (idx binIvfFlat) Params() map[string]string {
+	return map[string]string{
+		MetricTypeKey: string(idx.metricType),
+		IndexTypeKey:  string(IvfSQ8),
+		ivfNlistKey:   strconv.Itoa(idx.nlist),
+	}
+}
+
+func NewBinIvfFlat(metricType MetricType, nlist int) Index {
```
Comment (ThreadDao) [inline client/index/ivf.go:130]: Maybe NewBinIvfFlatIndex ?

File: client/index/ivf.go [hunk 2]
```diff
@@ -106,3 +110,30 @@ func NewIvfSQ8Index(metricType MetricType, nlist int) Index {
...
+	return ivfPQIndex{
```
Comment (ThreadDao) [inline client/index/ivf.go:131]: ivfPQIndex ?

File: client/index/ivf.go (hunk 1 above)
Comment (congqixia) [reply client/index/ivf.go]: fixed

File: client/index/ivf.go (hunk 2 above)
Comment (congqixia) [reply client/index/ivf.go]: fixed

**Review by ThreadDao — COMMENTED** [review]
/lgtm

**Pull request merged**